src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -557,29 +601,36 @@ class module_elf : public module_impl
         auto offset = rela->r_offset;
         size_t sec_size = 0;
         patcher::buf_type buf_type;
-        if (secname.compare(".ctrltext") == 0) {
-          sec_size = instrbuf.size();
+        if (secname.compare(Control_Code_Section_Name) == 0) {
+          sec_size = m_instr_buf.size();
           buf_type = patcher::buf_type::ctrltext;
         }
-        else if (secname.compare(".ctrldata") == 0) {
-          sec_size = ctrlpkt.size();
+        else if (m_ctrl_packet_exist && (secname.compare(Control_Packet_Section_Name) == 0)) {
+          sec_size = m_ctrl_packet.size();
           buf_type = patcher::buf_type::ctrldata;
         }
+        else if (m_save_buf_exist && (secname.compare(Preemtp_Save_Section_Name) == 0)) {
+            sec_size = m_save_buf.size();
+            buf_type = patcher::buf_type::preempt_save;
+        }
+        else if (m_restore_buf_exist && (secname.compare(Preempt_Restore_Section_Name) == 0)) {
+            sec_size = m_restore_buf.size();
+            buf_type = patcher::buf_type::preempt_restore;
+        }
```

Comment:
Are we trying to find what does this section/elf represent? Is yes, can you add a method like determine_section_type() add move this if/else chain to that function.